sw/host/opentitanlib/src/util/openocd.rs

```diff
@@ -0,0 +1,110 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{Context, Result};
+
+use std::fs::File;
+use std::io::Write;
+use std::process::{Command, Stdio, Child};
+use crate::io::jtag::{Jtag, JtagParams};
+
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+
+use std::cell::Cell;
+
+use std::net::TcpStream;
+
+use crate::impl_serializable_error;
+
+use std::{thread, time};
+
+
+/// Represents an OpenOCD server and we can interact with interactively
+pub struct OpenOcdClient {
+    /// OpenOCD child process
+    child: Child,
```

Comment:
nit, maybe better name?
```suggestion
    openocd_server_process: Child,
```

Reply:
Yes, will change that.